### PR TITLE
Add CHARRA and its dependencies

### PR DESCRIPTION
**Pull request by Asiderr**

### CODE REVIEW

File: recipes-core/qcbor/qcbor/0001-Makefile-Use-CC-as-a-compiler.patch [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+diff --git a/Makefile b/Makefile
+index 805e702..0bc386d 100644
+--- a/Makefile
++++ b/Makefile
+@@ -7,7 +7,7 @@
+ # See BSD-3-Clause license in README.md
+ #
+ 
+-CC=cc
++#CC=cc
```
Comment (macpijan) [inline recipes-core/qcbor/qcbor/0001-Makefile-Use-CC-as-a-compiler.patch]: CC?=cc should be enough to be able to override it through environment and not breaking the original Makefile

File: recipes-core/qcbor/qcbor_git.bb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+SRC_URI = "git://github.com/laurencelundblade/QCBOR.git;protocol=https;branch=master"
+SRC_URI += "file://0001-Makefile-Use-CC-as-a-compiler.patch"
+
+SRCREV = "c02e13eb6263381195290424c4cdffd440a0aa96"
+
+S = "${WORKDIR}/git"
+TARGET_CC_ARCH += "${LDFLAGS}"
+
+OE_EXTRAMAKE = "PREFIX=${exec_prefix}"
```
Comment (macpijan) [inline recipes-core/qcbor/qcbor_git.bb]: EXTRA_OEMAKE?

File: recipes-core/qcbor/qcbor/0001-Makefile-Use-CC-as-a-compiler.patch (hunk 1 above)
Comment (Asiderr) [reply recipes-core/qcbor/qcbor/0001-Makefile-Use-CC-as-a-compiler.patch]: Fixed

File: recipes-core/qcbor/qcbor_git.bb (hunk 2 above)
Comment (Asiderr) [reply recipes-core/qcbor/qcbor_git.bb]: Deleted

**Pull request merged**